common "raise" for py2/py3

erlichmen opened the pull request:
Under python3 throwing tuple is no longer supported, lucky python2 supports raise for re-thowring exceptions and maintaining the trackback.
Clean also spaces after EOL

jsatt commented:
Hey @erlichmen, thanks for the PR. Initially, it looks good. However, I'm trying to practical test and not able to produce any error like I suspect you may have been seeing based on the changes in the PR.Since the setup.py is using `use_2to3`, it's replacing the tuple with the proper raise under PY3 without your modifications.

Could you provide more info on the original problem? Python version? Maybe a snippet that produces the error?

erlichmen commented:
I might had an old version of fudge because when I modified the code I
notice the up to date code uses  tuple throw and I had
etype(val).with_traceback(..) when I modified the code.
Fell free to dump this PR if 2to3 handles this already although that this
PR uses the common method that AFAIK should work on py2 and py3 without any
2to3 involvement.

Also there is the end of line spaces cleanup (you might what to run pep8 in
the setup tool) but of course it is just bonus.

On Sat, Jan 27, 2018 at 11:51 PM, Jeremy Satterfield <
notifications@github.com> wrote:

> Hey @erlichmen <https://github.com/erlichmen>, thanks for the PR.
> Initially, it looks good. However, I'm trying to practical test and not
> able to produce any error like I suspect you may have been seeing based on
> the changes in the PR.Since the setup.py is using use_2to3, it's
> replacing the tuple with the proper raise under PY3 without your
> modifications.
>
> Could you provide more info on the original problem? Python version? Maybe
> a snippet that produces the error?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fudge-py/fudge/pull/7#issuecomment-361018117>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAqOb0VEtwLUVyQmtt7CBgRAAeJQLh3jks5tO5psgaJpZM4RrHBA>
> .
>

jsatt commented:
@erlichmen, since there's no immediate issue, let's hold off on this for now. Feel free to reopen if you have an example of the issue happening in an installed state.
